crates/cast/src/main.rs

```diff
@@ -84,44 +95,45 @@ async fn main() -> Result<()> {
     update_cast_config(&mut config, &cli);
 
     let provider = get_provider(&config.rpc_url)?;
+    let mut account: Option<SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>> = None;
+
+    match &cli.command {
+        Commands::Call(_) | Commands::Account(_) => {}
+        Commands::Multicall(value)
+            if matches!(
+                &value.command,
+                starknet_commands::multicall::Commands::New(_)
+            ) => {}
+        _ => account = Some(get_account(&cli, &config, &provider).await?),
+    }
```

Comment:
Removing repetition, account is being fetched in one place only now

Reply:
I suppose, but now we have two match statements that basically do the same, except the first one sets up an account if commands are different than call or multicall new. This is especially confusing since the second match actually executes the commands, so the first one seems rather redundant. I think that getting an account in places it is needed, in only one match statement (as it was previously), in this case can greatly improve the code readability and maintainability. I know the `get_account` will be repeated, but I think this is a small price + we can get rid of `expects` to the accounts. Does that make sense?